### PR TITLE
Bugfix/ena metadata

### PR DESCRIPTION
**Pull request by tgurbich**
Added changes to improve location metadata handling for isolates. Added an alternative lookup from API if the XML data retrieval fails and accounted for different possible attribute names of geolocation field in JSON.

### CODE REVIEW

File: bin/create_metadata_table.py [hunk 1]
```diff
@@ -128,27 +128,17 @@ def get_metadata(acc):
...
         pass
     elif acc.startswith("GCA"):
-        json_data_gca = load_xml(acc)
-        biosample = json_data_gca["ASSEMBLY_SET"]["ASSEMBLY"]["SAMPLE_REF"]["IDENTIFIERS"]["PRIMARY_ID"]
-        project = json_data_gca["ASSEMBLY_SET"]["ASSEMBLY"]["STUDY_REF"]["IDENTIFIERS"]["PRIMARY_ID"]
+        try:
+            json_data_gca = load_xml(acc)
+            biosample = json_data_gca["ASSEMBLY_SET"]["ASSEMBLY"]["SAMPLE_REF"]["IDENTIFIERS"]["PRIMARY_ID"]
+            project = json_data_gca["ASSEMBLY_SET"]["ASSEMBLY"]["STUDY_REF"]["IDENTIFIERS"]["PRIMARY_ID"]
+        except:
```
Comment (mberacochea) [inline bin/create_metadata_table.py:135]: We should capture the exception error mesasge in the log file

File: bin/create_metadata_table.py [hunk 2]
```diff
@@ -128,27 +128,17 @@ def get_metadata(acc):
...
+            try:
+                biosample, project = ena_api_request(acc)
+            except:
```
Comment (mberacochea) [inline bin/create_metadata_table.py:138]:
```suggestion
            except:
                # logging.exception will capture the exception information automatically 
                logging.exception("Could not obtain biosample and project information for {}".format(acc)")
```

File: bin/get_ENA_metadata.py [hunk 3]
```diff
@@ -59,12 +59,23 @@ def get_location(sample_id):
...
+    if not json_data:
+        return None
+    
+    possible_keys = ["characteristics.geo loc name", "characteristics.geo_loc_name", "description.geo_loc_name"]
+    for key_pair in possible_keys:
+        try:
+            attr1, attr2 = key_pair.split('.')
+            geo_data_list = json_data[attr1][attr2]
+            break
+        except KeyError:
```
Comment (mberacochea) [inline bin/get_ENA_metadata.py:74]: Should we print this as a warning?

**Review by mberacochea — CHANGES_REQUESTED** [review]
Thanks Tanya!... I've added some minor comments, just to print some extra logging info which may help in the future

**Comment by tgurbich** [comment]
Thank you, @mberacochea. Added logging.

**Review by mberacochea — APPROVED** [review]

**Pull request merged**